geps/gep-1713/index.md

```diff
@@ -0,0 +1,408 @@
+
+TBD: Can conditional defaulting be done with CEL?
+
+#### GatewaySpec.Listeners MinItems
+
+TBD: Do we want to make this change?
+
+Currently when creating a Gateway you must specify at least a single listener. With `ListenerSets` this opens the possibility of wanting to create a Gateway with no listeners.
+
+#### Route Attaching
+
+Routes MUST be able to specify a `ListenerSet` as a `parentRef` and make use of the `sectionName` field in `ParentReference` to help target a specific listener. If no listener is targeted (`sectionName`/`port` are unset) then the Route references all the listeners on the `ListenerSet`. It `MUST NOT` attach to additional listeners on the parent `Gateway`.
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: HTTPRoute
+metadata:
+  name: httproute-example
+spec:
+  parentRefs:
+  - name: second-workload-listeners
+    kind: ListenerSet
+    sectionName: second
+```
+
+For instance, the following `HTTPRoute` attemps to attach to a listener defined in the parent `Gateway`. This is not valid and the route's status should reflect that.
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: HTTPRoute
+metadata:
+  name: httproute-example
+spec:
+  parentRefs:
+  - name: some-workload-listeners
+    kind: ListenerSet
+    sectionName: foo
+```
+
+#### Listener Validation
+
+Implementations MUST treat the parent `Gateway` as having the concatenated list of all listeners from itself and attached `ListenerSets`
+Validation of this list of listeners MUST behave the same as if the list were part of a single `Gateway`.
+
+From the earlier example the above resources would be equivalent to a single `Gateway` where the listeners are collapsed into a single list.
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1
+kind: Gateway
+metadata:
+  name: parent-gateway
+spec:
+  gatewayClassName: example
+  listeners:
+  - name: foo
+    hostname: foo.com
+    protocol: HTTP
+    port: 80
+  - name: first
+    hostname: first.foo.com
+    protocol: HTTPS
+    port: 443
+    tls:
+      mode: Terminate
+      certificateRefs:
+      - kind: Secret
+        group: ""
+        name: first-workload-cert # Provisioned via HTTP01 challenge
+  - name: second
+    hostname: second.foo.com
+    protocol: HTTPS
+    port: 443
+    tls:
+      mode: Terminate
+      certificateRefs:
+      - kind: Secret
+        group: ""
+        name: second-workload-cert # Provisioned via HTTP01 challenge
+```
+
+#### Listener Precedence
+
+Listeners should be merged using the following precedence:
+1. "parent" Gateway
+2. ListenerSet ordered by creation time (oldest first)
+3. ListenerSet ordered alphabetically by “{namespace}/{name}”.
+
+If there are listener conflicts, this should be reported as `Conflicted=True` in the ListenerStatus as usual. See 'Conditions' section below for more details on object should report the conflict.
```

Comment:
Which resources would this condition appear on? The parent Gateway? Only ListenerSets with a conflicted Listener? All connected resources that contain Listeners?

Reply:
> Only ListenerSets with a conflicted Listener

I updated the text to clarify this